components/org.wso2.bfsi.consent.management.extensions/src/main/java/org/wso2/bfsi/consent/management/extensions/admin/impl/DefaultConsentAdminHandler.java

```diff
@@ -0,0 +1,335 @@
+
+        //retrieve the total of the data set queried
+        if (limit != null || offset != null) {
+            try {
+                ArrayList<DetailedConsentResource> results = ConsentExtensionsDataHolder.getInstance()
+                        .getConsentCoreService().searchDetailedConsents(consentIDs,
+                        clientIDs, consentTypes, consentStatuses, userIDs, fromTime, toTime, null, null);
+                total = results.size();
+            } catch (ConsentManagementException e) {
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+            }
+        }
+
+        JSONObject metadata = new JSONObject();
+        metadata.put(ConsentExtensionConstants.COUNT, count);
+        metadata.put(ConsentExtensionConstants.OFFSET, offset);
+        metadata.put(ConsentExtensionConstants.LIMIT, limit);
+        metadata.put(ConsentExtensionConstants.TOTAL, total);
+
+        response.put(ConsentExtensionConstants.METADATA, metadata);
+        consentAdminData.setResponseStatus(ResponseStatus.OK);
+        consentAdminData.setResponsePayload(response);
+    }
+
+    @Override
+    public void handleRevoke(ConsentAdminData consentAdminData) throws ConsentException {
+
+        try {
+            Map queryParams = consentAdminData.getQueryParams();
+
+            String consentId = ConsentAdminUtils.validateAndGetQueryParam(queryParams,
+                    ConsentExtensionConstants.CC_CONSENT_ID);
+            if (consentId == null) {
+                throw new ConsentException(ResponseStatus.BAD_REQUEST, "Mandatory parameter consent ID not available");
+            } else {
+                ConsentResource consentResource = ConsentExtensionsDataHolder.getInstance().getConsentCoreService()
+                        .getConsent(consentId, false);
+
+                if (!ConsentExtensionConstants.AUTHORIZED_STATUS.equals(consentResource.getCurrentStatus())) {
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST,
+                            "Consent is not in a revocable status");
+                } else {
+                    ConsentExtensionsDataHolder.getInstance().getConsentCoreService()
+                            .revokeConsentWithReason(ConsentAdminUtils.validateAndGetQueryParam(queryParams,
+                                    ConsentExtensionConstants.CC_CONSENT_ID),
+                                    ConsentExtensionConstants.REVOKED_STATUS,
+                                    ConsentAdminUtils.validateAndGetQueryParam(queryParams, "userID"),
+                                    ConsentExtensionConstants.CONSENT_REVOKE_FROM_DASHBOARD_REASON);
+                }
+            }
+            consentAdminData.setResponseStatus(ResponseStatus.OK);
+            consentAdminData.setResponseStatus(ResponseStatus.NO_CONTENT);
+        } catch (ConsentManagementException e) {
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
```

Comment:
Let's add throwable object to the execption.

Reply:
Fixed in https://github.com/wso2-extensions/bfsi-consent-management/pull/9/commits/7640a3fd126e5e00ce693961fdc6c5cd28816bb9